runtime/src/accounts_db.rs

```diff
@@ -775,14 +775,20 @@ impl AccountsDB {
             }
         }
 
-        for account_infos in purges.values() {
+        for (pubkey, account_infos) in &purges {
             let mut no_delete = false;
+            let mut would_unref_count = 0;
             for (_slot, account_info) in account_infos {
                 if *store_counts.get(&account_info.store_id).unwrap() != 0 {
                     no_delete = true;
                     break;
+                } else {
+                    would_unref_count += 1;
                 }
             }
+            if would_unref_count != accounts_index.ref_count_from_storage(&pubkey) {
```

Comment:
[nits] Basically `would_unref_count` is same as `account_infos.len()`, so I think we could simplify this further. (Well, the original code should have done so... by me #8724; In other words, no need to count in fact; but i was too slanted to the concept of ref-COUNT-ing... xD):

```rust
if account_infos.len() != accounts_index.ref_count_from_storage(&pubkey) {
    no_delete = true;
} else {
    for (_slot, account_info) in account_infos {
        if *store_counts.get(&account_info.store_id).unwrap() != 0 {
            no_delete = true;
            break;
        }
    }
}
```

Reply:
Yea, that logic is better, I'll change to that. Thanks!